src/components/FormSelectInput.component.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import styled from "styled-components";
+
+import { defaultTheme } from "../theme";
+import { InputStyle } from "./styled";
+
+type MapElement = { id: number; name: string };
+interface State extends MapElement {
+  cities: Array<MapElement>;
+}
+export const FormSelectInput = ({
+  label,
+  register,
+  onFocus,
+  onBlur,
+  onChange,
+  loading,
+  data,
+  disabled,
+}: {
+  label: string;
+  register: any;
+  onFocus?: any;
+  onBlur?: any;
+  onChange?: (e: React.ChangeEvent) => void;
+  loading: boolean;
+  data: any;
+  disabled?: boolean;
+}) => {
+  return (
+    <Select
+      {...register}
+      onFocus={onFocus}
+      onChange={onChange ? onChange : register.onChange}
+      disabled={disabled}
+      defaultValue=""
+      onBlur={onBlur ? onBlur : register.onBlur}
+    >
+      <option value="">{loading ? "Loading..." : "Select an option"}</option>
+      {label === "Cities" ? (
+        !loading &&
+        data?.map((state: State) => (
```

Comment:
why do you think you need to add the `?` to the `data`? can you think on a different way to improve this code?

Reply:
I think the intellisense autocomplete put that there, and I know it makes sense, because I'm not type safing data to have elements, the way I think this code could improve is by adding this type safe to data, I mean, adding a fast return when data is empty.